docs/architecture/how/resharding.md

```diff
@@ -47,61 +66,85 @@ StateSync in this phase would send the ``StateSplitRequest`` to the ``SyncJobsAc
-
 
 ### Why delayed receipts are special?
 For all the other columns, there is no dependency between entries, but in case of delayed receipts - we are forming a 'queue'. We store the information about the first index and the last index (in DelayedReceiptIndices struct).
 
-Then, we receipt arrives, we add it as the 'DELAYED_RECEIPT + last_index' key (and increment last_index by 1).
+Then, when receipt arrives, we add it as the 'DELAYED_RECEIPT + last_index' key (and increment last_index by 1).
+
+That is why we cannot move this trie entry type in the same way as others where account id is part of the key. Instead we do it by iterating over this queue and inserting entries to the queue of the relevant child shard. 
+
+
+## Constraints
+
+The state sync of the parent shard, the resharing and the catchup of the children shards must all complete within a single epoch. 
+
+
+## Rollout
+
+### Flow
+
+The resharding will be initiated by having it included in a dedicated protocol version together with neard . Here is the expected flow of events:
+
+* A new neard release is published and protocol version upgrade date is set to D, roughly a week from the release. 
+* All node operatores upgrade their binaries to the newly released version within the given timeframe, ideally as soon as possible but no later than D. 
+* The protocol version upgrade voting takes place at D in an epoch E and nodes vote in favour of switching to the new protocol version in epoch E+2. 
+* The resharding begins at the beginning of epoch E+1. 
+* The network switches to the new shard layout in the first block of epoch E+2. 
+
 
-That's why we cannot move it column 'in trie parts' (like we did for others), but we do it by 'iterating' over this queue.
+### Monitoring
 
+Resharding exposes a number of metrics and logs that allow for monitoring the resharding process as it is happening. Resharding requires manual recovery in case anything goes wrong and should be monitored in order to ensure smooth node operation. 
 
+* near_resharding_status is the primary metric that should be used for tracking the progress of resharding. It's tagged with a shard_uid label of the parent shard. It's set to corresponding ReshardingStatus enum and can take one of the following values
+  * 0 - Scheduled - resharding is scheduled and waiting to be executed. 
+  * 1 - Building - resharding is running. Only one shard at a time can be in that state while the rest will be either finished or waiting in the Scheduled state. 
+  * 2 - Finished - resharding is finished. 
+  * -1 - Failed - resharding failed and manual recovery action is required. The node will operate as usual until the end of the epoch but will then stop being able to process blocks. 
+* near_resharding_batch_size and near_resharding_batch_count - those two metrics show how much data has been resharded. Both should be gradually increasing while the near_resharding_status. 
```

Comment:
Can split `near_resharding_batch_size` and `near_resharding_batch_count` across two points

Reply:
nit: Both should be gradually increasing while the near_resharding_status -> Both metrics should progress with the `near_resharding_status` as follows

Reply:
Can include graphics to show what this might look like. I've taken screenshots from grafana dashboard

![Screenshot 2023-11-29 at 6 52 12 PM](https://github.com/near/nearcore/assets/16697830/7dc5e1f4-d27c-4132-9ee7-b33788bcc2ed)
![Screenshot 2023-11-29 at 6 52 04 PM](https://github.com/near/nearcore/assets/16697830/8c0c5963-fdf4-432c-800e-6c01ab8e15d0)